File: Palindrome numbers - GFG/palindrome-numbers.java
//{ Driver Code Starts
//Initial Template for Java


import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        Long t = Long.parseLong(read.readLine());
        while(t-- > 0)
        {
            long N = Long.parseLong(read.readLine());
            Solution ob = new Solution();
            System.out.println(ob.isPallindrome(N));
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution{
    static long isPallindrome(long n){
     String val = Long.toBinaryString(n); 
     
     int left=0;
     int right=val.length()-1;
     
     while(left<right){
         if(val.charAt(left)!=val.charAt(right)){
             return 0;
         }
         left++;right--;
     }
     
     return 1;

    }
}

File: Palindrome numbers - GFG/palindrome-numbers.js
//{ Driver Code Starts
//Initial Template for javascript


'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let N = parseInt(readLine());
        let obj = new Solution();
        let res = obj.isPallindrome(N);
        console.log(res);
    }
}


// } Driver Code Ends


//User function Template for javascript

/**
 * @param {number} N
 * @return {number}
*/

class Solution {
    isPallindrome(n){
        let x=n.toString(2);
        let mod=0;
        let rev=0;
        let res=0;
        let temp=x;
        while(x!=0){
            mod=x%10;
            rev=rev*10+mod;
            x=parseInt(x/10)
        }
        
          if(temp==rev){
            if(temp<0)return 0;
            return 1;
        }else{
            return 0;
        }
        
    }
}